### DISCUSSION TITLE
Yubi key registration does not work.

### DISCUSSION THREAD

**Discussion by piotrowy** [dani-garcia/vaultwarden#3159] [Q&A]
### Subject of the issue

Yubi key registration does not work

### Deployment environment

* vaultwarden version: 2022.12.0

* Install method: On dedicated server from source

* Clients used: Desktop

* Reverse proxy and version: none

* MySQL/MariaDB or PostgreSQL version: nope

* Other relevant details:

### Steps to reproduce
I tried to register yubikey series 5. It says invalid otp. I reset otp on my ubikey and I still see the same error message.

### Expected behaviour
Yubikey is registered.

### Actual behaviour
It always says "invalid otp".

### Troubleshooting data
<img width="322" alt="obraz" src="https://user-images.githubusercontent.com/7810642/213880595-1d074843-41bb-4924-a492-bf8cfbc7f1ed.png">

**Comment by juliazero** [comment]
@piotrowy  Did you follow wiki to enable OTP support?
https://github.com/dani-garcia/vaultwarden/wiki/Enabling-Yubikey-OTP-authentication

**Reply by piotrowy** [reply]
I'm gonna try it. Thanks!

**Comment by BlackDex** [comment]
Follow that guide. Or use WebAuthn.
I tested it just recently and it works fine after the registration and providing the api keys.

**Comment by RainSun0322** [comment]
Today I met the same error. I have been using Yubikey OTP for months, but when I tried to log in today, the yubikey OTP page kept loading after I touched my key. I had to go to the admin page to disable yubikey 2FA. Then, I tried to re-personalize my yubikey and registered it on upload.yubico.com. Then, I went to https://upgrade.yubico.com/getapikey/ to get a new api key and replaced the values in the admin page. However, when trying to add the yubikey OTP again, it kept saying 'Invalid yubikey OTP provided' after about 25 seconds of loading. The erro happened when using 1.27.0, but I have upgraded to the latest version 1.28.1 and tried again. The error was the same. @BlackDex

**Reply by BlackDex** [reply]
Please enable debug logging, try to register the key again and provide the logs generated during that time.

**Reply by RainSun0322** [reply]
> Please enable debug logging, try to register the key again and provide the logs generated during that time.

I tried again and it seemed working correctly. Maybe there was some temporary connection errors between my server and yubico server. Thanks for your reply.
